### ISSUE TITLE
Cinnamenu (Cinnamenu@json): Cinnamenu glitches

### ISSUE THREAD

**Issue by mfreeman72** [linuxmint/cinnamon-spices-applets#4444]
```
 * Applet version/Build date: Version 5.0.1 (2022-08-19 14:14:57)
 * Cinnamon version: 5.4.11
 * Distribution: Linux Mint 21
 * Graphics hardware *and* driver used: AMD Ryzen 7/Vega 7 APU running on included open source Mesa drivers
 
 ```

**Notify author of applet**
@fredcw

**Issue**
Recently (since the previous Cinnamenu update, I think), there have been visual and functional glitches in Cinnamenu. When scrolling, small pieces (1 or more pixels wide) get sheered off of the right-most edge of the fonts in the icon names and jumps vertically up and down randomly around the menu space. Also, if I move my mouse cursor just below the menu (about where the next couple rows of icons would be if the menu were extended downward below where it's actually drawn), I can see tool-tip descriptions for icons that are below the currently-displayed menu space, and if I click on the desktop below the menu, it will run the application whose icon is described in the tool-tip.

**Steps to reproduce**
This happens always with this version of the menu, so just place it on the panel, open it, scroll the menu up and down, and move/click the mouse around the desktop area just below the menu.

**Expected behaviour**
These glitches shouldn't happen.

**Other information**

**Comment by fredcw** [comment]
Thanks for reporting this. I'm not getting the missing pixels or the menu jumping but am getting the invisible part of the scroll window being clickable outside of the menu. This bug is also present in the default cinnamon menu applet. Unfortunately, I don't know how to fix this or what the cause is. It would be better to file a bug report in linuxmint/cinnamon as this is a cinnamon issue.

**Comment by fredcw** [comment]
I have opened an issue in cinnamon: https://github.com/linuxmint/cinnamon/issues/11123

**Comment by mfreeman72** [comment]
Here's an example of the pixel issue I'm seeing. I can't take an actual screenshot of it because as soon as I make the computer do anything (mouse moves, keypresses, etc.), the pixel issue corrects itself. So this is a picture shot with my cell phone:

![Bad Pixels in Menu](https://user-images.githubusercontent.com/7842664/187503465-842318c2-f4a3-42d9-9d26-d3438759d521.jpg)

As you can see, a few pixels' worth of the icon title's font gets stuck when I scroll, specifically if the icon title is particularly long. And sometimes, these groups of pixels jump upward and downward between two different vertical positions on the menu. I'm not sure if this is related to the issue you reported in the cinnamon issue you opened or not. I've only seen this happen recently, and at the same time as that issue. If you'd like, I can add it to that issue as well, or if you think it's a different or unrelated issue, I can just leave it here.

**Comment by fredcw** [comment]
@mfreeman72 The invisible menu items outside of the menu problem seems to disappear in cinnamon 5.4.12 Are the other issues still present?

**Comment by mfreeman72** [comment]
That strange. For me, the invisible menu items outside of the menu still exist on Cinnamon 5.4.12. And yes, the font issue still exists as well.

**Comment by fredcw** [comment]
Are you using Mint 21 daily PPA or another OS?

**Comment by mfreeman72** [comment]
I'm using just the standard Mint 21, installed from the official release ISO.

**Comment by fredcw** [comment]
You're right sorry, it's still there. I've reopened the cinnamon issue.

**Comment by fredcw** [comment]
It's strange that nobody else has mentioned this. It's quite a big bug and yet I can't find any other references to it, especially as it affects the default cinnamon menu as well.

**Comment by mfreeman72** [comment]
Agreed. It's a pretty obnoxious bug, too. I've accidentally ran several programs I never wanted to, just by trying to close the window.

**Comment by fredcw** [comment]
I forgot about the glitch bug until I noticed it again recently myself. I think this is a glitch in the underlying GTK library so I don't there's much I can do about it.

**Comment by mfreeman72** [comment]
Yes, agreed. It's also happening with the normal built-in Cinnamon menu, not just Cinnamenu. I hope it gets some attention from Cinnamon developers. I'm inadvertently opening programs all the time because of this bug. Quite frustrating!

**Comment by fredcw** [comment]
No I meant the stray pixels at the side of the buttons. The buttons have integer widths which means that depending on the width of the menu, there could be a small gap at the end. GTK doesn't seem to be erasing the contents of this gap properly. Carefully adjusting the width of the menu can get rid of the gap but that's not really a fix.

**Comment by mfreeman72** [comment]
Ok. I see. I've just been living with those, trying to ignore them. I'll have to experiment with the menu width and see if I can get those glitches minimized better. I feel like there have been more little annoying bugs like this (and others) in this Linux Mint version than we've seen before, or at least in a long time.
